ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -276,10 +278,13 @@ private Response executeStatements(
   private <T extends Statement> KsqlEntity executeStatement(
       final PreparedStatement<T> statement,
       final Map<String, Object> propertyOverrides,
-      final KsqlEntityList entities) {
+      final KsqlEntityList entities
+  ) {
     try {
       final Handler<T> handler = getCustomExecutor(statement);
       if (handler != null) {
+        waitForPreviousDistributedStatementToBeHandled(entities);
```

Comment:
is this applicable to everything that has a custom handler (e.g. RunScript/TerminateQuery)? Since this is blocking behavior, we may want to err on the side of enumerating the cases that need to be waited for.

Reply:
RunScript and TerminateQuery most definitely.  There are a couple we _could_ choose to not do this for, e.g. SetProperty, DescribeFunction.

I'll pick this up in my next PR.